### PR TITLE
Implement side effects when domain gets approved from django admin

### PR DESCRIPTION
**Pull request by rachidatecs**
## 🗣 Description ##

- Define all status transitions in application admin class, 
- Implement approved email notification, 
- Refactor email methods in application model class from DRYer code, 
- Unit test emails as side-effects of status change and admin save,
- Unit tests against transitions that are not allowed

## 💭 Motivation and context ##

- Solves [422](https://github.com/cisagov/getgov/issues/442) and [722](https://github.com/cisagov/getgov/issues/722)
- Establishes infrastructure for further work on Domain Application status changes and flows

### CODE REVIEW

**Review by abroddrick — COMMENTED** [review]
Posting first batch of comments

File: src/registrar/admin.py [hunk 1]
```diff
@@ -90,16 +90,36 @@ def save_model(self, request, obj, form, change):
...
-                # This is a transition annotated method in model which will throw an
-                # error if the condition is violated. To make this work, we need to
-                # call it on the original object which has the right status value,
-                # but pass the current object which contains the up-to-date data
-                # for the email.
-                original_obj.in_review(obj)
+            if obj.status != original_obj.status:
+                if obj.status == models.DomainApplication.STARTED:
+                    # No conditions
+                    pass
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/admin.py:96]: Correct

File: src/registrar/admin.py [hunk 2]
```diff
@@ -90,16 +90,36 @@ def save_model(self, request, obj, form, change):
...
+                if obj.status == models.DomainApplication.SUBMITTED:
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/admin.py:97]: Good point, I'll revise

**Review by abroddrick — APPROVED** [review]
looks great! Tested on my end

**Pull request merged**